sdk/lib/rtl/amd64/dynfntbl.c

```diff
@@ -0,0 +1,323 @@
+/*
+ * PROJECT:     ReactOS RTL
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Dynamic function table support routines
+ * COPYRIGHT:   Copyright 2022 Timo Kreuzer (timo.kreuzer@reactos.org)
+ */
+
+#include <rtl.h>
+
+#define NDEBUG
+#include <debug.h>
+
+#define TAG_RTLDYNFNTBL 'tfDP'
+
+typedef
+_Function_class_(GET_RUNTIME_FUNCTION_CALLBACK)
+PRUNTIME_FUNCTION
+GET_RUNTIME_FUNCTION_CALLBACK (
+    _In_ DWORD64 ControlPc,
+    _In_opt_ PVOID Context
+    );
+typedef GET_RUNTIME_FUNCTION_CALLBACK *PGET_RUNTIME_FUNCTION_CALLBACK;
+
+typedef
+_Function_class_(OUT_OF_PROCESS_FUNCTION_TABLE_CALLBACK)
+DWORD   
+OUT_OF_PROCESS_FUNCTION_TABLE_CALLBACK (
+    _In_ HANDLE Process,
+    _In_ PVOID TableAddress,
+    _Out_ PDWORD Entries,
+    _Out_ PRUNTIME_FUNCTION* Functions
+    );
+typedef OUT_OF_PROCESS_FUNCTION_TABLE_CALLBACK *POUT_OF_PROCESS_FUNCTION_TABLE_CALLBACK;
+
+typedef enum _FUNCTION_TABLE_TYPE
+{
+    RF_SORTED = 0x0,
+    RF_UNSORTED = 0x1,
+    RF_CALLBACK = 0x2,
+    RF_KERNEL_DYNAMIC = 0x3,
+} FUNCTION_TABLE_TYPE;
+
+typedef struct _DYNAMIC_FUNCTION_TABLE
+{
+    LIST_ENTRY ListEntry;
+    PRUNTIME_FUNCTION FunctionTable;
+    LARGE_INTEGER TimeStamp;
+    ULONG64 MinimumAddress;
+    ULONG64 MaximumAddress;
+    ULONG64 BaseAddress;
+    PGET_RUNTIME_FUNCTION_CALLBACK Callback;
+    PVOID Context;
+    PWCHAR OutOfProcessCallbackDll;
+    FUNCTION_TABLE_TYPE Type;
+    ULONG EntryCount;
+    //RTL_BALANCED_NODE TreeNodeMin;
+    //RTL_BALANCED_NODE TreeNodeMax;
```

Comment:
Why are these commented out?
Process Hacker has one `RTL_BALANCED_NODE` here.

Reply:
I have it from Windows 11 symbols. The reason it's commented out is that rtl uses a precompiled header and it's impossible to include ntdef into it.